### PR TITLE
add recipe for dav1d

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jaimergp** [comment]
@conda-forge/help-c-cpp This is ready for review, thanks!

File: recipes/dav1d/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  sha256: 047b8229511a82b5718a1d34c86c067b078efd02f602986d2ed09b23182ec136
+
+build:
+  number: 0
+  script:
+    - set -ex
+    - >
+        meson setup _build
+        ${MESON_ARGS:---prefix=${PREFIX} --libdir=lib}  # [unix]
+        --prefix=%LIBRARY_PREFIX%  # [win]
```
Comment (ocefpaf) [inline recipes/dav1d/meta.yaml:23]: Nice trick!

File: recipes/dav1d/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://code.videolan.org/videolan/dav1d/-/archive/{{ version }}/dav1d-{{ version }}.tar.gz
+  sha256: 047b8229511a82b5718a1d34c86c067b078efd02f602986d2ed09b23182ec136
+
+build:
+  number: 0
```
Comment (carterbox) [inline recipes/dav1d/meta.yaml:17]: Missing a run_export!

File: recipes/dav1d/meta.yaml (hunk 1 above)
Comment (jaimergp) [reply recipes/dav1d/meta.yaml:23]: Saw it in another feedstock using Meson!

File: recipes/dav1d/meta.yaml (hunk 2 above)
Comment (jaimergp) [reply recipes/dav1d/meta.yaml:17]: Argh! I'll add it on the feedstock.

**Pull request merged**